### ISSUE TITLE
Unexpected deprecation message related to an entity

### ISSUE THREAD

**Issue by javiereguiluz** [symfony/symfony#24569]
| Q                | A
| ---------------- | -----
| Bug report?      | yes
| Feature request? | no
| BC Break report? | no
| RFC?             | no
| Symfony version  | 3.4

We're updating the Symfony Demo application to Symfony 3.4 (see https://github.com/symfony/symfony-demo/pull/673). Most of the triggered deprecations make sense, but we're struggling with this one:

![entity-deprecation](https://user-images.githubusercontent.com/73419/31599863-75b4214c-b254-11e7-92a4-8f528b33de46.png)

Why does that deprecation occur? We even exclude the entire `Entity/` directory from autowiring:

```yaml
    App\:
        resource: '../src/*'
        exclude: '../src/{DataFixtures,Entity,Repository,Tests}'
```

**Comment by stof** [comment]
@javiereguiluz I think this is related to the autowiring of controller arguments (where you might have a typehint for a Post relying on param converters). As there is no autowirable service for this typehint, the AutowirePass triggers the BC layer to try finding one such service by type.

**Comment by javiereguiluz** [comment]
@stof you are right! We use `Post` as a "param converter" in lots of controllers like this one:

```php
public function showAction(Post $post): Response
```

Could this be considered a bug? We want to use `Post` as a param converter, not as a service injection.

**Comment by iltar** [comment]
Pretty much anything could be added as arguments there via the new controller arguments event and argument value resolvers. Besides of this, pretty much anything that's being set in the request object can be injected this way, not sure if you can catch all those cases.

**Comment by yceruto** [comment]
This process is carried out in `AutowirePass` and `RegisterControllerArgumentLocatorsPass`, is there a way to know all the excluded resources there? maybe we could verify it to omit these cases?

**Comment by iltar** [comment]
@yceruto no, you can't know this as a lot of things are resolved during run-time execution. You can only know if they are available as service or not.

**Comment by yceruto** [comment]
So if we have explicit configuration that excludes resources:
```yaml
App\:
    resource: '../src/*'
    exclude: '../src/{Entity,Repository,Tests}' # <------
```
we can't get this information into `*Pass` to avoid auto-registration of controller typed arguments that aren't available as service?
```xml
<service id="autowired.App\Entity\Post" class="App\Entity\Post" public="false" autowire="true"/>
```
this service is defined at compilation-time, so I hope there are some solution for that.

**Comment by weaverryan** [comment]
So this is one of the big reasons that we removed the auto-registration of services for autowiring in 4.0. So in 4.0, it’s not an issue - the entities are excluded in services.yml, and then are not auto-registered during autowiring.

But yes, the deprecation warning is a problem - a bug that we need to fix somehow.

**Comment by Tobion** [comment]
This is basically fixed with #24671 when enabled, isn't it?

**Comment by kbond** [comment]
#24671 solves this problem for me.
